src/Web/HttpApiData/Internal/FormUrlEncoded.hs

```diff
@@ -0,0 +1,163 @@
+{-# LANGUAGE CPP                        #-}
+{-# LANGUAGE DefaultSignatures          #-}
+{-# LANGUAGE DeriveGeneric              #-}
+{-# LANGUAGE FlexibleContexts           #-}
+{-# LANGUAGE FlexibleInstances          #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE KindSignatures             #-}
+{-# LANGUAGE OverloadedStrings          #-}
+{-# LANGUAGE PolyKinds                  #-}
+{-# LANGUAGE ScopedTypeVariables        #-}
+{-# LANGUAGE TypeFamilies               #-}
+{-# LANGUAGE TypeOperators              #-}
+module Web.HttpApiData.Internal.FormUrlEncoded where
+
+#if __GLASGOW_HASKELL__ < 710
+import Control.Applicative
+#endif
+
+import           Control.Arrow           (first, left, second)
+import           Control.Monad.State
+import qualified Data.ByteString.Lazy    as BSL
+import qualified Data.Map                as M
+import           Data.Monoid
+import           Data.String.Conversions (cs)
+import qualified Data.Text               as T
+import           GHC.Exts                (IsList (..))
+import           GHC.Generics
+import           Network.URI             (escapeURIString, isUnreserved,
+                                          unEscapeString)
+
+import Web.HttpApiData.Internal.HttpApiData
+
+-- | The contents of a form, not yet url-encoded.
+newtype Form = Form { unForm :: M.Map T.Text T.Text }
```

Comment:
Is there a reason to use lazy `Map` and not strict `Map` or `HashMap`? /cc @jkarni